practices/multiple_environments.md

```diff
@@ -0,0 +1,125 @@
+2. **Testing environment**
+   The purpose of the test environment is to allow human testers to exercise new and changed code via either automated checks or non-automated techniques. After the developer accepts the new code and configurations through unit testing in the development environment, the items are moved to one or more test environments. Upon test failure, the test environment can remove the faulty code from the test platforms, contact the responsible developer, and provide detailed test and result logs. 
+   If all tests pass, the test environment or a continuous integration framework controlling the tests can automatically promote the code to the next deployment environment.
+3. **Staging environment**
+   The staging environment or pre-production environment is an environment for testing that exactly resembles a production environment. It seeks to mirror an actual production environment as closely as possible and may connect to other production services and data, such as databases. 
+   The primary use of a staging environment is to test all the installation/configuration/migration scripts and procedures before they're applied to a production environment. This ensures all major and minor upgrades to a production environment are completed reliably, without errors, and in a minimum of time.
+   Another important use of staging is performance testing, particularly load testing, as this is often sensitive to the environment.
+   Staging environmet is also used by some organizations to preview new features to select customers or to validate integrations with live versions of external dependencies.
+3. **Production environment**
+   The production environment can also be said as a live environment, particularly for servers, as it is the environment that users directly interact with. Deploying to production is the most sensitive step; it may be done by deploying new code directly, or by deploying a configuration change.
+   Deploying a new release generally requires a restart, unless hot swapping is possible, and thus requires either an interruption in service (usual for user software, where applications are restarted) or redundancy – either restarting instances slowly behind a load balancer or starting up new servers ahead of time and then simply redirecting traffic to the new servers.
+   
+
+![Multiple Environments](/files/Multiple_Environments.png)</br>
+Source: [Using Multiple Environments to Improve Your Development Workflow](https://deploybot.com/blog/using-multiple-environments-to-improve-your-development-workflow)
+
+## Why Use Multi Environments
+
+Multi Environments helps you with:
+
+- **Keeps the team productive**
+  Having multiple environments enables a team to work on parallel development efforts. If there are several people working on the app, using a different environment for each group of people helps keep the team productive.
```

Comment:
Please, rewrite this with your own words, it is too similar to the original:

 _Having multiple environments enables a team to work on parallel development efforts. If there are several people working on the app, using a different environment for each group of people helps keep the team productive._